### DISCUSSION TITLE
Beatmaps missing after lazer reinstall

### DISCUSSION THREAD

**Discussion by tsbreuer** [ppy/osu#15207] [Q&A] [answered]
I have an issue where after reinstalling lazer and making it change to its old directory, its not seemingly working.

i think the problem might be here, its not allowing me to set the directory to the existing directory, is there a way to tell lazer to recognizing an existing directory?

Heres the error:
![imagen](https://user-images.githubusercontent.com/6902643/138158692-cf298365-f128-427d-a341-009e82db285e.png)

**Comment by bdach** [comment, marked as answer]
the error is telling you that you cannot move your *fresh* lazer install to where your files *used* to be. because it would overwrite your old files.

as requested on discord, please provide screenshots of your `%appdata%/osu` folder, screenshots of your old `files` folder that isn't getting detected, contents of `%appdata%/storage.ini`, as well as game logs.

**Comment by tsbreuer** [comment]
appdata:
![imagen](https://user-images.githubusercontent.com/6902643/138159717-4d9d253e-1339-4f40-a2bc-98771e1a6382.png)

Old folder data:
![imagen](https://user-images.githubusercontent.com/6902643/138159464-650a02b5-044a-4431-9b86-10a5e6fc1735.png)

Storage.ini:
![imagen](https://user-images.githubusercontent.com/6902643/138159762-ab370dcd-652a-4f81-be38-bc041afe1d4e.png)

**Reply by bdach** [reply]
you've put the `storage.ini` file in *local* appdata, and not roaming. there's a difference, lazer executables go in the former, and user data goes in the latter. please enter `%appdata%/osu` into a shell/cmd/explorer and move that `storage.ini` file there.

**Reply by tsbreuer** [reply]
That worked. I wonder if its worth considering a "locate existing install" next to the "change folder" button? Would make this a lot easier and less confusing.

**Reply by bdach** [reply]
if you're asking for a button to open the roaming appdata folder then this already exists.

![2021-10-20-214107_620x663_scrot](https://user-images.githubusercontent.com/20418176/138160885-a4e698cf-e7e6-4881-a2e0-d7cb366c4ed7.png)

if you're asking for a button that would be able to find your _old_ install wherever it might be then that's arguably out of scope, because due to allowing users to migrate to arbitrary locations we would have to scan your entire filesystem to look for it.

**Reply by tsbreuer** [reply]
Second option is what i was thinking of, but i dont mean to scan the entire filesystem. 

Just like how its detecting if existing files are there before migrating, you could tell lazer to check the specific folder for those files, and move there.

**Reply by mtsoltan** [reply]
I have the same problem. Here is my osu config folder:
![image](https://user-images.githubusercontent.com/39483620/193161850-494bf5ae-57ad-45e4-8e45-b7b84de0366f.png)

The old installation used to use this folder, and I assumed the new one will take over just fine, only to figure out it required me to re-login, and none of my beatmaps were being recognized.

Should I open a new question for it, or keep it in this discussion for completeness?

**Reply by peppy** [reply]
You are migrating from a VERY old install. We only supported migration from the old database version for 8 months, and have since removed it. To migrate correctly, you have two options:

First, before doing anything, create a backup of `client.db` and keep it safe in case anything goes wrong

1:

- Delete client.realm
- Install this version of osu! https://github.com/ppy/osu/releases/tag/2022.911.0. Do not uninstall first.
- Wait for it to migrate your `client.db` to `client.realm`
- Let it update to the latest release

2:

- Zip your `client.db` and attach it here or email to me at pe@ppy.sh. I'll upgrade it for you.

**Reply by mtsoltan** [reply]
Done. Instructions from option 1 work perfectly.

Thank you very much!
